### PR TITLE
Fix incorrect layout for point scales

### LINKED ISSUE #1807: Incorrect Positioning for Vertical Point scales
vega_editor

### PR DESCRIPTION
**Pull request by kanitw**
Fix #1807

The output look good now!

vega_editor

vega_editor

### CODE REVIEW

File: test/compile/layout.test.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ describe('compile/layout', () => {
       });
 
       const sizeExpr = unitSizeExpr(model, X);
-      assert.equal(sizeExpr, 'max(datum["distinct_a"] + 2*0.5, 0) * 21');
+      assert.equal(sizeExpr, 'max(datum["distinct_a"] - 1 + 2*0.5, 0) * 21');
```
Comment (domoritz) [inline test/compile/layout.test.ts]: -1+1 = 0

Comment (kanitw) [inline test/compile/layout.test.ts]: Sure, but 0.5 can vary based on paddingOuter. So there isn't a better.

Comment (kanitw) [inline test/compile/layout.test.ts]: Note that I intentionally chose 0.5 as the default because - 1 + 2*0.5 = 0

Comment (domoritz) [inline test/compile/layout.test.ts]: I guess I would like to see a comment that explains that 0.5 is the passingOuter.

Comment (kanitw) [inline test/compile/layout.test.ts]: I'm NOT gonna add comments to unit tests unless it's necessary.  
This is a fairly small method, you can easily read the code which is already well commented. 

Plus, I'm not changing the 0.5 part in this PR.

**Pull request merged**